### PR TITLE
* Buttons

### PR DESCRIPTION
**Pull request by Wagnerp**

### CODE REVIEW

File: Source/Krypton Toolkit/Examples/Buttons/Buttons.csproj [hunk 1]
```diff
@@ -1,10 +1,22 @@
...
 
   <PropertyGroup>
-    <TargetFramework>net5.0-windows</TargetFramework>
+    <TargetFramework>net48</TargetFramework>
     <LangVersion>latest</LangVersion>
     <OutputType>WinExe</OutputType>
     <UseWindowsForms>true</UseWindowsForms>
   </PropertyGroup>
 
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
```
Comment (Smurf-IV) [inline Source/Krypton Toolkit/Examples/Buttons/Buttons.csproj:10]: Why is this here, Buttons should not rely on "Bitness"

File: Source/Krypton Toolkit/Main/Krypton.Toolkit.Suite.Extended.Buttons/Controls Toolkit/Dialog Buttons/KryptonAbortDialogButton.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
+﻿using Krypton.Toolkit.Suite.Extended.Language.Model;
```
Comment (Smurf-IV) [inline Source/Krypton Toolkit/Main/Krypton.Toolkit.Suite.Extended.Buttons/Controls Toolkit/Dialog Buttons/KryptonAbortDialogButton.cs:1]: Can you change your sorting order so that System usings are first, then a gap , then others are included. 
- There is a reason to this madness: I have found in the past, that the C# Compiler caches the '`System.* usings` first, so when compiling large projects, just this simple thing can improve the throughput of the processing.

File: Source/Krypton Toolkit/Main/Krypton.Toolkit.Suite.Extended.Buttons/Controls Toolkit/Dialog Buttons/KryptonAbortDialogButton.cs [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+            ParentChanged += KryptonAbortDialogButton_ParentChanged;
+
+            TextChanged += KryptonAbortDialogButton_TextChanged;
+        }
+
+        private void KryptonAbortDialogButton_TextChanged(object sender, EventArgs e)
+        {
+            if (Text == Name)
+            {
+                Text = "&Abort";
```
Comment (Smurf-IV) [inline Source/Krypton Toolkit/Main/Krypton.Toolkit.Suite.Extended.Buttons/Controls Toolkit/Dialog Buttons/KryptonAbortDialogButton.cs:35]: If this is not to be Translated, then put an `@` before the first `"`

File: Source/Krypton Toolkit/Main/Krypton.Toolkit.Suite.Extended.Buttons/Controls Toolkit/Dialog Buttons/KryptonCancelDialogButton.cs [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    {
+        private KryptonForm _parent;
+
+        public KryptonForm ParentWindow { get => _parent; set { _parent = value; Invalidate(); } }
+
+        public KryptonCancelDialogButton()
+        {
+            DialogResult = DialogResult.Cancel;
+
+            ParentChanged += KryptonCancelDialogButton_ParentChanged;
```
Comment (Smurf-IV) [inline Source/Krypton Toolkit/Main/Krypton.Toolkit.Suite.Extended.Buttons/Controls Toolkit/Dialog Buttons/KryptonCancelDialogButton.cs:18]: I have found it is best practice to put the equivalent release `-=` of these "events" ina dispose: i.e.
`ParentChanged -= KryptonCancelDialogButton_ParentChanged;`

File: Source/Krypton Toolkit/Shared/Krypton.Toolkit.Suite.Extended.Language.Model/Classes/ButtonLanguageController.cs [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+        /// <param name="language">The language.</param>
+        /// <param name="okText">The ok text.</param>
+        /// <param name="yesText">The yes text.</param>
+        /// <param name="noText">The no text.</param>
+        /// <param name="cancelText">The cancel text.</param>
+        /// <param name="abortText">The abort text.</param>
+        /// <param name="ignoreText">The ignore text.</param>
+        private static void SetLanguage(SelectedLanguage language, string okText = "", string yesText = "", string noText = "", string cancelText = "", string abortText = "", string ignoreText = "")
+        {
+            switch (language)
```
Comment (Smurf-IV) [inline Source/Krypton Toolkit/Shared/Krypton.Toolkit.Suite.Extended.Language.Model/Classes/ButtonLanguageController.cs:28]: Should these also have the equivalent shortcut chars set as well ?

**Review by Smurf-IV — CHANGES_REQUESTED** [review]

File: Source/Krypton Toolkit/Main/Krypton.Toolkit.Suite.Extended.Buttons/Controls Toolkit/Dialog Buttons/KryptonAbortDialogButton.cs (hunk 3 above)
Comment (Smurf-IV) [reply Source/Krypton Toolkit/Main/Krypton.Toolkit.Suite.Extended.Buttons/Controls Toolkit/Dialog Buttons/KryptonAbortDialogButton.cs:35]: This can be applied to all text's

Comment (Wagnerp) [reply Source/Krypton Toolkit/Main/Krypton.Toolkit.Suite.Extended.Buttons/Controls Toolkit/Dialog Buttons/KryptonAbortDialogButton.cs:35]: I need to do more work on this, i.e. if a language has been changed, then grab the equivalent from a string array.

File: Source/Krypton Toolkit/Main/Krypton.Toolkit.Suite.Extended.Buttons/Controls Toolkit/Dialog Buttons/KryptonAbortDialogButton.cs (hunk 2 above)
Comment (Wagnerp) [reply Source/Krypton Toolkit/Main/Krypton.Toolkit.Suite.Extended.Buttons/Controls Toolkit/Dialog Buttons/KryptonAbortDialogButton.cs:1]: I'll turn off automatic sorting of usings

File: Source/Krypton Toolkit/Main/Krypton.Toolkit.Suite.Extended.Buttons/Controls Toolkit/Dialog Buttons/KryptonAbortDialogButton.cs [hunk 6]
```diff
@@ -17,13 +17,45 @@ public class KryptonAbortDialogButton : KryptonButton
...
+        #endregion
+
         #region Properties
-        public SelectedLanguage SelectedLanguage { get => _selectedLanguage; set { _selectedLanguage = value; Invalidate(); } }
+        public SelectedLanguage SelectedLanguage
+        {
+            get => _selectedLanguage;
+
+            set
+            {
```
Comment (Smurf-IV) [inline Source/Krypton Toolkit/Main/Krypton.Toolkit.Suite.Extended.Buttons/Controls Toolkit/Dialog Buttons/KryptonAbortDialogButton.cs:43]: For Event firing `sets` you should check to see if the value has changed 1st

**Comment by Smurf-IV** [comment]
@Wagnerp 
Still needs changes, and conflicts sorting

**Comment by Wagnerp** [comment]
> @Wagnerp
> Still needs changes, and conflicts sorting

@Smurf-IV Will look at it this week

File: Source/Krypton Toolkit/Shared/Krypton.Toolkit.Suite.Extended.Language.Model/Classes/ButtonLanguageController.cs (hunk 5 above)
Comment (Wagnerp) [reply Source/Krypton Toolkit/Shared/Krypton.Toolkit.Suite.Extended.Language.Model/Classes/ButtonLanguageController.cs:32]: I don't think so, as the optional parameters are there for other languages that are not supported by the language manager.

**Comment by Wagnerp** [comment]
Have to redo the buttons module, as it was a mess!

File: Source/Krypton Toolkit/Main/Krypton.Toolkit.Suite.Extended.Buttons/Controls Toolkit/Dialog Buttons/KryptonAbortDialogButton.cs [hunk 7]
```diff
@@ -35,22 +45,30 @@ public class KryptonAbortDialogButton : KryptonButton
...
+        /// <summary>Gets or sets the selected language.</summary>
+        /// <value>The selected language.</value>
         public SelectedLanguage SelectedLanguage
         {
             get => _selectedLanguage;
 
             set
             {
-                SelectedLanguageChangedEventArgs e = new SelectedLanguageChangedEventArgs(value);
+                if (_selectedLanguage == value)
```
Comment (Smurf-IV) [inline Source/Krypton Toolkit/Main/Krypton.Toolkit.Suite.Extended.Buttons/Controls Toolkit/Dialog Buttons/KryptonAbortDialogButton.cs]: This should be `!=`

File: Source/Krypton Toolkit/Main/Krypton.Toolkit.Suite.Extended.Buttons/Controls Toolkit/Dialog Buttons/KryptonCancelDialogButton.cs [hunk 8]
```diff
@@ -54,5 +62,12 @@ protected override void OnPaint(PaintEventArgs e)
 
             base.OnPaint(e);
         }
+
+        ~KryptonCancelDialogButton()
+        {
+            GC.SuppressFinalize(this);
```
Comment (Smurf-IV) [inline Source/Krypton Toolkit/Main/Krypton.Toolkit.Suite.Extended.Buttons/Controls Toolkit/Dialog Buttons/KryptonCancelDialogButton.cs]: I thought these where supposed to be in the `Dispose(bool)` functions

**Comment by Wagnerp** [comment]
We now have working UAC buttons!

**Comment by Smurf-IV** [comment]
Cool.. Now just need the readme.md and an updated picture please?

**Comment by Wagnerp** [comment]
> Cool.. Now just need the readme.md and an updated picture please?

Not yet, will be implementing the other button types over this week.

**Comment by Wagnerp** [comment]
Something is wrong with `KryptonCommandLinkVersion2`, can't seem to pinpoint it. It's not behaving like this: https://github.com/Krypton-Suite/Extended-Toolkit/blob/master/Assets/Examples/CommandLinks.png?raw=true

**Comment by Wagnerp** [comment]
> Something is wrong with `KryptonCommandLinkVersion2`, can't seem to pinpoint it. It's not behaving like this: 

Figured it out

**Comment by Wagnerp** [comment]
Good to go... for now! :)
